Sources/edge/Commands/Devices.swift

```diff
@@ -0,0 +1,38 @@
+import ArgumentParser
+import Foundation
+import Logging
+
+protocol DeviceDiscovery {
+    func listUSBDevices(logger: Logger) async
+    func listEthernetInterfaces(logger: Logger) async
```

Comment:
Can we make these functions return their output as a struct? That way we can serialize the data as JSON down the line using something like `--format json`. Would be helpful to have

Reply:
I'm actually toying around with that but I'm in a dirty state because I'm also trying to use `libusb` and wrap it instead of `lsusb`. Mind if I do json output in the next PR?